app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java

```diff
@@ -515,7 +515,22 @@ protected void setupDownloadOptions() {
         videoButton.setVisibility(isVideoStreamsAvailable ? View.VISIBLE : View.GONE);
         subtitleButton.setVisibility(isSubtitleStreamsAvailable ? View.VISIBLE : View.GONE);
 
-        if (isVideoStreamsAvailable) {
+        prefs = PreferenceManager.getDefaultSharedPreferences(getContext());
+        String defaultMedia = prefs.getString(getString(R.string.last_used_download_type),
+                    getString(R.string.video));
+
+        assert defaultMedia != null;
+        if (isVideoStreamsAvailable && (defaultMedia.equals(getString(R.string.video)))) {
```

Comment:
`(defaultMedia.equals(getString(R.string.video))` this will fail to detect the last used type when the user changed tha pplication language. Please create three keys in settings_keys.xml to avoid that.